### PR TITLE
[MRG+1] Exclude to layout.

### PR DESCRIPTION
**Pull request by jaeilepp**
Closes https://github.com/mne-tools/mne-python/issues/3618

I suspect something is going to break...

### CODE REVIEW

**Comment by jaeilepp** [comment]
I didn't find anything going wrong with this. Circle failure is unrelated.

**Comment by agramfort** [comment]
I just restarted circle. Let's see...

**Comment by Eric89GXL** [comment]
+1 for merge

Looks like it's hitting the circle failure problem we saw before on your branches. @jaeilepp I wonder if it's a problem with it caching the source repo. We might need to force it to update, and do our comparison of changed files to upstream/master, which I don't think it's doing now. I'll have a look.

File: mne/channels/layout.py [hunk 1]
```diff
@@ -456,6 +456,15 @@ def find_layout(info, ch_type=None, exclude='bads'):
...
         layout.names = _clean_names(layout.names, before_dash=True)
 
+    # Apply mask for excluded channels.
+    if exclude == 'bads':
+        exclude = info['bads']
+    mask = np.ones(len(layout.names), dtype=bool)
+    mask[[layout.names.index(c) for c in exclude if c in layout.names]] = False
+    layout.names = list(np.array(layout.names)[mask])
+    layout.pos = layout.pos[mask]
+    layout.ids = list(np.array(layout.ids)[mask])
```
Comment (agramfort) [inline mne/channels/layout.py]: +1

**Comment by agramfort** [comment]
thx @jaeilepp

**Pull request merged**